fees/manager.go

```diff
@@ -231,53 +231,45 @@ func computeNextPriceWindow(
 		start := slot * consts.Uint64Len
 		window.Update(&newRollupWindow, start, previousConsumed)
 	}
-	total := window.Sum(newRollupWindow)
 
-	nextPrice := previousPrice
-	if total > target {
+	var (
+		totalUnitsConsumed = window.Sum(newRollupWindow)
+		nextUnitFee        = previousUnitFee
+	)
+	switch {
+	case totalUnitsConsumed == target:
+		return nextUnitFee, newRollupWindow, nil
+	case totalUnitsConsumed > target:
 		// If the parent block used more units than its target, the baseFee should increase.
-		delta := total - target
-		x := previousPrice * delta
-		y := x / target
-		baseDelta := y / changeDenom
-		if baseDelta < 1 {
-			baseDelta = 1
-		}
-		n, over := math.Add64(nextPrice, baseDelta)
+		rawDelta := previousUnitFee * (totalUnitsConsumed - target) / target
+		delta := max(rawDelta/changeDenom, 1) * changeDenom // price must change in increments on changeDenom
```

Comment:
This is the main point: `delta` here is ensured to be a multiple of `changeDenom`. Moverover, from a dimensional analysis standpoint, delta is a "token amont" quantity, which can be added to unit fees.
In the previous version `baseDelta` is a dimensionless quantity (ration of two fees) which cannot be added to a dimensioned quantity like next price

Reply:
This is actually wrong: this code comes from coreth (see https://github.com/ava-labs/coreth/blob/58729bccbbda5fc2ffe4edc1d635d81bb2437421/consensus/dummy/dynamic_fees.go#L40) which in turns comes from EIP1559 proposal (see for instance https://github.com/ethereum/EIPs/issues/1559)

The goal of changeDenom is **not** to make sure that delta happens in multiples of the changeDenom.
Instead the idea is to cap the unit fees changes to ±1/changeDenom (1/8 = 12.5% in eth proposal).